modules/javafx.graphics/src/main/native-prism-d3d/D3DMeshView.cc

```diff
@@ -154,19 +159,41 @@ void D3DMeshView::render() {
         return;
     }
 
-    float lightsColor[12];
-    for (int i = 0, j = 0; i < 3; i++) {
+    float lightsColor[12];       // 3 lights x (3 color + 1 padding)
+    float lightsAttenuation[12]; // 3 lights x (3 attenuation factors + 1 padding)
+    float lightsRange[12];		 // 3 lights x (1 maxRange + 3 padding)
+    for (int i = 0, c = 0, a = 0, r = 0; i < 3; i++) {
         float w = lights[i].w;
-        lightsColor[j++] = lights[i].color[0] * w;
-        lightsColor[j++] = lights[i].color[1] * w;
-        lightsColor[j++] = lights[i].color[2] * w;
-        lightsColor[j++] = 1;
+        lightsColor[c++] = lights[i].color[0] * w;
+        lightsColor[c++] = lights[i].color[1] * w;
+        lightsColor[c++] = lights[i].color[2] * w;
+        lightsColor[c++] = 1;
+
+        lightsAttenuation[a++] = lights[i].attenuation[0];
+        lightsAttenuation[a++] = lights[i].attenuation[1];
+        lightsAttenuation[a++] = lights[i].attenuation[2];
+        lightsAttenuation[a++] = 0;
```

Comment:
In the initial commit this filed was used for `maxRange`. I think that was good idea. Was there any issue with that approach ?

Reply:
`maxRange` has its own array to reserve space for `minRange` if we ever do that (see [this discussion](https://github.com/javafxports/openjdk-jfx/issues/256#issuecomment-469970201)), so I future-proofed it.